integration_test.py

```diff
@@ -0,0 +1,131 @@
+## test with python -m pytest integration_test.py
+
+import os
+from testutils import simplecpp, format_include_path_arg, format_include
+
+def __test_relative_header_create_header(dir, with_pragma_once=True):
+    header_file = os.path.join(dir, 'test.h')
+    with open(header_file, 'wt') as f:
+        f.write(f"""
+                {"#pragma once" if with_pragma_once else ""}
+                #ifndef TEST_H_INCLUDED
+                #define TEST_H_INCLUDED
+                #else
+                #error header_was_already_included
+                #endif
+                """)
+    return header_file, "error: #error header_was_already_included"
+
+def __test_relative_header_create_source(dir, include1, include2, is_include1_sys=False, is_include2_sys=False):
+    src_file = os.path.join(dir, 'test.c')
+    with open(src_file, 'wt') as f:
+        f.write(f"""
+                #undef TEST_H_INCLUDED
+                #include {format_include(include1, is_include1_sys)}
+                #include {format_include(include2, is_include2_sys)}
+                """)
+    return src_file
+
+
+def test_relative_header_0_rel(tmpdir):
```

Comment:
it feels like these tests should be parametrized.

Here is an example where a parameter "builddir" is created and pytest will run the test function both when it is False and True:
https://github.com/danmar/cppcheck/blob/main/test/cli/whole-program_test.py#L215

Reply:
Done. Added more coverage this way too